routes/ecom/modules/list-payments.js

```diff
@@ -3,12 +3,40 @@
 // generate default payment gateway object
 const newPaymentGateway = require(process.cwd() + '/lib/new-payment-gateway')
 
+const checkDiscount = (paymentGateway, items) => {
+  const discount = paymentGateway.discount
+  const productIds = discount?.product_ids
+  if (items && items.length) {
+    if (discount.apply_at !== 'freight' && discount.value > 0 && productIds && productIds.length) {
```

Comment:
```suggestion
    if (discount.apply_at === 'subtotal' && discount.value > 0 && productIds && productIds.length) {
```

Já que aqui você trata o desconto apenas no subtotal...